src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,84 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
+    public int calculateIndex(Object key, int tableCapacity) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % tableCapacity;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= this.table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = calculateIndex(key, table.length);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+        } else {
```

Comment:
```suggestion
        if (table[index] == null) {
            table[index] = new Node<>(key, value);
            size++;
            return;
        } 
```
the more there are levels of indentations, the harder it's to read